source/features/collapsible-content-button.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+Insert collapsible content when writing comments (via `<summary>`)
+
+https://user-images.githubusercontent.com/1402241/53678019-0c721280-3cf4-11e9-9c24-4d11a697f67c.png
+*/
+
+import React from 'dom-chef';
+import select from 'select-dom';
+import onetime from 'onetime';
+import delegate from 'delegate';
+import features from '../libs/features';
+import * as icons from '../libs/icons';
+
+const addEvents = onetime(() => {
+	delegate('.rgh-collapsible-content-btn', 'click', addContentToDetails);
+});
+
+// Wraps string in at least 2 \n on each side,
+// as long as the field doesn't already have them.
+// Code adopted/adapted from GitHub.
+function smartBlockWrap(content: string, field: HTMLTextAreaElement) {
+	const before = field.value.slice(0, field.selectionStart);
+	const after = field.value.slice(field.selectionEnd);
+	const [whitespaceAtStart] = before.match(/\n*$/) || [''];
+	const [whitespaceAtEnd] = after.match(/^\n*/) || [''];
+	let newlinesToAppend = '';
+	let newlinesToPrepend = '';
+	if (before.match(/\S/) && whitespaceAtStart.length < 2) {
+		newlinesToPrepend = '\n'.repeat(2 - whitespaceAtStart.length);
+	}
+
+	if (after.match(/\S/) && whitespaceAtEnd.length < 2) {
+		newlinesToAppend = '\n'.repeat(2 - whitespaceAtEnd.length);
+	}
+
+	return newlinesToPrepend + content + newlinesToAppend;
+}
```

Comment:
Wow, this is a really cool way to dynamically determine the amount of whitespace to lead/trail with!